### ISSUE TITLE
OWA: Too many connections, and 100% CPU when searching for persons

### ISSUE THREAD

**Issue by benbucksch** [mustang-im/mustang#150]
### Reproduction
1. Setup a few mail accounts with a realistic amount of email and contacts
2. Open the Vertical view
![image](https://github.com/user-attachments/assets/bf7e88d0-b28b-42f7-83c2-c7d0b4aa1533)
3. Switch to Person view
![image](https://github.com/user-attachments/assets/78fe201b-d7ab-4e27-9425-6747eb3d710b)
4. Click on 1-2 persons

### Actual result
* CPU fan spins up, with 100% CPU, for multiple minutes
* Error: "Error: Too many concurrent connections opened., Cannot open mailbox"
```
Error: Too many concurrent connections opened., Cannot open mailbox. Server = ....EURP191.PROD.OUTLOOK.COM, mdbGuid = ..., maiboxGuid = ...
  at OWAAccount.callOWA (/logic/Mail/OWA/OWAAccount.ts:136:13)
  at async OWAEMail.markRead (/logic/Mail/OWA/OWAEMail.ts:91:5)
```
* The display updates every 30 to 60 seconds
* The Mustang window keeps grabbing focus away from other windows, e.g. the browser tab that I'm typing into here, interfering with the bug report filing.
* Mustang is essentially locked up since 5-10 minutes, and I cannot even close it.

### Expected result
* No server calls
* A single SQL query matching the person selected
* Results display immediately
* Low CPU usage, and only for a split-second

**Comment by NeilRashbrook** [comment]
I don't make server calls if the UI doesn't ask for them. The OWA code is very similar to the IMAP code in that it unconditionally makes a server call every time the UI marks a message as read. I assume you're getting away with this in IMAP because the library requires you to lock the mailbox effectively serialising the requests?

I have no idea why your UI might be trying to grab focus. is it doing this in response to trying to display the error banner?

**Comment by benbucksch** [comment]
Did you try the reproduction? Make sure to use a fresh @outlook.com account that you don't need.

**Comment by NeilRashbrook** [comment]
I tried the reproduction with the Office 365 account but I didn't even hit my breakpoint on `callOWA` once, so your STR are insufficient.

**Comment by benbucksch** [comment]
There are 2 problems here:
* The UI is triggering too many requests at the same time. I know it's caused by the Persons search, because I observed it, but I don't know where and how exactly. That's to be debugged.
* Our EWS client is overloading the server. That's #121 .

**Comment by benbucksch** [comment]
I did. Corrected.

**Comment by benbucksch** [comment]
Fixed in git commit 997f5b19
